pandas/tests/indexes/multi/test_conversion.py

```diff
@@ -47,7 +47,12 @@ def test_array_interface(idx):
         return
 
     # for MultiIndex, copy=False is never allowed
-    with pytest.raises(ValueError, match="Unable to avoid copy while creating"):
+    msg = "Starting on NumPy 2.0, the behavior of the 'copy' keyword has changed "
+    "and passing 'copy=False' raises an error when a zero-copy NumPy array "
+    "is not possible, Pandas will follow this behavior starting with "
+    "version 3.0. This conversion to NumPy requires a copy, but "
+    "'copy=False' was passed. Consider using 'np.asarray(..)' instead."
+    with tm.assert_produces_warning(FutureWarning, match=msg):
```

Comment:
It's fine to just test a small part of the message, just so we assert the correct message is raised. Something like

```
msg = "Pandas will follow this behavior starting with version 3.0"
with tm.assert_produces_warning(FutureWarning, match=msg):
    ...
```

Reply:
And for the other CI errors, essentially you will have to replace all occurrences of `with pytest.raises(ValueError, match="Unable to avoid copy while creating")` with this `assert_produces_warning`

Reply:
Thanks, will work on this in a couple of days. Been too focused on AoC this December 😆

Reply:
is it alright if I match the first part of the message? 
```python
    msg = "Starting with NumPy 2.0, the behavior of the 'copy' keyword"
    with tm.assert_produces_warning(FutureWarning, match=msg):
        np.array(arr, copy=False)
```
Using 
```python
msg = "Pandas will follow this behavior starting with version 3.0"
```
is failing